src/interceptors/ClientRequest/utils/recordRawHeaders.test.ts

```diff
@@ -241,3 +241,15 @@ it('does not throw on using Headers before recording', () => {
   request.headers.set('X-My-Header', '1')
   expect(getRawFetchHeaders(request.headers)).toEqual([['X-My-Header', '1']])
 })
+
+it('does not use the same instance of rawHeaders', async () => {
```

Comment:
`rawHeaders` inclusion in the name is probably not needed. How about

```ts
it('isolates the headers on the instance basis')
```

Reply:
I also like linking GitHub issues above the test case names so it's easier to grasp the context for bug fixes.